provider/core.rb

```diff
@@ -70,14 +62,28 @@ class FileTemplate
     attr_accessor :async
     attr_accessor :resource_name
 
-    def initialize(options)
-      options.each { |k, v| method("#{k}=").call(v) }
+    class << self
+      # Construct a new FileTemplate based on a resource object
+      def file_for_resource(output_folder, object, version, config, env)
+        file_template = new(output_folder, object.name, object.__product, version, env)
+        file_template.object = object
+        file_template.config = config
+        file_template
+      end
 
-      @env = {
-        pyformat_enabled: options.dig(:env, :pyformat_enabled),
-        goformat_enabled: options.dig(:env, :goformat_enabled),
-        start_time: options.dig(:env, :start_time)
-      }
+      # Construct a new FileTemplate based on a file to be copied into the output directory
+      def file_for_copy(output_folder, target, api, version, env)
```

Comment:
It looks like this is a dead method (and isn't necessary since it just calls the constructor directly)

Reply:
Ah yeah, missed taking this back out. Good catch!